signimage/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM registry.fedoraproject.org/fedora as ksource
+RUN yum install -y kernel-devel
+RUN cp /usr/src/kernels/*/scripts/sign-file /sign-file
+
+FROM golang:1.18 as builder
+
+WORKDIR /workspace
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+COPY signimage.go signimage.go
+# Build
+RUN go build signimage.go
+
+FROM registry.fedoraproject.org/fedora
+
+COPY --from=builder /workspace/signimage /
+COPY --from=ksource /sign-file /sign-file
```

Comment:
Any reason we are not using the binary from that layer instead of coping it from the previous identical layer?

Reply:
we could install kernel-devel in this layer, but its huge (80MB plus dependencies) and we only need sign-file (15760 bytes) from it, so we do it this way to try and keep the resultant image as small as possible.

Reply:
can we just use any image, and not fedora? for example alpine?

Reply:
its a little trickier than that, in that sign-file needs libcrypto, glibc etc, whiich is why the first and third layers are the same base image. In theory we could move to alpine for both,  I'm using fedora because it avoid a bunch of repo and entitlement related issues I had when using ubi-minimal. I dont have enough knowledge of alpine to get sign-image and its deps installed reliably, but theres nothing to stop it being changed in the future.